deepdoc/parser/json_parser.py

```
@@ -33,6 +33,14 @@ def __init__(self, max_chunk_size: int = 2000, min_chunk_size: int | None = None
     def __call__(self, binary):
         encoding = find_codec(binary)
         txt = binary.decode(encoding, errors="ignore")
+        # Strip a leading UTF-8 BOM (U+FEFF). Windows editors, PowerShell
+        # `Out-File` and several .NET export paths write JSON with a BOM by
+        # default. The bytes decode cleanly under UTF-8 but ``json.loads``
+        # rejects the leading \ufeff as garbage, and the bare ``except`` in
+        # ``_parse_json`` / ``_parse_jsonl`` swallows the failure so the
+        # whole upload silently produces zero chunks (#19179).
+        if txt.startswith("\ufeff"):
+            txt = txt[1:]
```

Ruff (0.16.3)

[error] Prefer `str.removeprefix()` over conditionally replacing with slice.

Use removeprefix instead of assignment conditional upon startswith.

(FURB188)